src/Microsoft.DotNet.Tools.Publish/Program.cs

```diff
@@ -89,6 +112,64 @@ private static bool CheckArg(CommandOption argument)
             return true;
         }
 
+        private static bool IsUnsupportedFramework(string framework)
+        {
+            return NuGetFramework.Parse(framework).Equals(NuGetFramework.UnsupportedFramework);
+        }
+
+        private static string GetProjectDirectory(string project)
+        {
+            return project.ToLower().EndsWith(Project.FileName) ?
+                                    Path.GetDirectoryName(project) :
+                                    project;
+        }
+
+        private static string GetProjectLockFile(string project)
+        {
+            string projectDir = GetProjectDirectory(project);
+            return Path.Combine(projectDir, LockFile.FileName);
+        }
+
+        // return the matching framework/runtime targets in the lock file
+        // if 'framework' or 'runtime' is null or empty then it matches with any.
+        private static IEnumerable<LockFileTarget> GetMatchingTargets(LockFile lockFile, string framework, string runtime)
```

Comment:
This would be a good helper to consolidate into the `LockFile` class itself. Something like `public IEnumerable<LockFileTarget> GetMatchingTargets(NuGetFramework framework, string runtimeIdentifier)`.